test/keygen_user_test.dart

```diff
@@ -0,0 +1,288 @@
+      );
+
+    });
+
+    setUp(() async {
+
+      user = await usersApi.createUser(
+        token: adminToken,
+        firstName: firstNameJohn,
+        email: emailJohnDoe,
+        password: passwordJohnDoe,
+      );
+
+      await Future.delayed(rateLimitDelay);
+
+    });
+
+    tearDown(() async {
+
+      await Future.delayed(rateLimitDelay);
+
+      bool valid = await usersApi.deleteUser(
+        token: adminToken,
+        user: user,
+      );
+
+      expect(valid, true);
+
+    });
+
+    // Updates the specified user resource by setting the values of the
+    // parameters passed.
+    //
+    // https://keygen.sh/docs/api/users/#users-update
+    test('update user', () async {
+
+      //
+      // first verify that user has firstNameJohn
+      // then change user name
+      // then verify that user now has firstNameBob
+      //
+
+      User user2 = await usersApi.retrieveUser(
+        token: adminToken,
+        userEmail: emailJohnDoe,
+      );
+
+      await Future.delayed(rateLimitDelay);
+
+      expect(user2.attributes.firstName, firstNameJohn);
+
+      await usersApi.updateUser(
+        token: adminToken,
+        user: user2,
+        firstName: firstNameBob,
+      );
+
+      await Future.delayed(rateLimitDelay);
+
+      user2 = await usersApi.retrieveUser(
+        token: adminToken,
+        userEmail: emailJohnDoe,
+      );
+
+      expect(user2.attributes.firstName, firstNameBob);
+
+    });
+
+    // Returns a list of users.
+    //
+    // https://keygen.sh/docs/api/users/#users-list
+    test('list users', () async {
+
+      List<User> users = await usersApi.listUsers(
+        token: adminToken,
+      );
+
+      expect(users.length, 1);
+      expect(users[0].id, user.id);
+
+    });
+
+    // Updates the user's password.
+    //
+    // https://keygen.sh/docs/api/users/#users-actions-update-password
+    test('update password as admin', () async {
+
+      try {
+
+        await usersApi.updateUserPassword(
+          token: adminToken,
+          user: user,
+          currentPassword: passwordJohnDoe,
+          newPassword: newPasswordJohnDoe,
+        );
+
+        fail('should have thrown');
+
+      } on KeygenError catch (e) {
+
+        expect(e.e?.code, 403);
+      }
+
+    });
+
+    // Updates the user's password.
+    //
+    // https://keygen.sh/docs/api/users/#users-actions-update-password
+    test('update password as user', () async {
+
+      Token userToken = await usersApi.createUserToken(
+        token: adminToken,
+        user: user,
+      );
+
+      await Future.delayed(rateLimitDelay);
+
+      try {
+
+        await usersApi.updateUserPassword(
+          token: userToken,
+          user: user,
+          currentPassword: passwordJohnDoe,
+          newPassword: newPasswordJohnDoe,
+        );
+
+        await Future.delayed(rateLimitDelay);
+
+        //
+        // cannot get new password
+        // just test that actions succeeds and does not throw
```

Comment:
solid, saw the note on the docs to use the update user endpoint to truely update the user's password